ivy_tests/test_ivy/test_frontends/test_tensorflow/test_tensor.py

```diff
@@ -70,10 +73,266 @@ def _check_query(query):
+    assert ta.dtype == ta_frontend.dtype.ivy_dtype
+
+
+@given(l_kwargs=_helper_random_tensorarray(fn="gather"))
+def test_tensorarray_gather(
+    l_kwargs,
+    backend_fw,
+):
+    ta, ta_frontend = _helper_init_tensorarray(backend_fw, l_kwargs[:2])
+    *_, indices = l_kwargs
+    helpers.value_test(
+        ret_np_from_gt_flat=ta.gather(indices).numpy().flatten(),
+        ret_np_flat=np.array(ta_frontend.gather(indices)).flatten(),
+        backend=backend_fw,
+    )
+
+
+@given(l_kwargs=_helper_random_tensorarray())
+def test_tensorarray_read(
+    l_kwargs,
+    backend_fw,
+):
+    ta, ta_frontend = _helper_init_tensorarray(backend_fw, l_kwargs)
+    id_read, _ = l_kwargs
+    for id, read in id_read:
+        helpers.value_test(
+            ret_np_from_gt_flat=ta.read(id).numpy().flatten(),
+            ret_np_flat=np.array(ta_frontend.read(id)).flatten(),
+            backend=backend_fw,
+        )
+
+
+@given(l_kwargs=_helper_random_tensorarray(fn="scatter"))
+def test_tensorarray_scatter(
+    l_kwargs,
+    backend_fw,
+):
+    id_read, _ = l_kwargs
+    ta, ta_frontend = _helper_init_tensorarray(backend_fw, l_kwargs, "scatter")
+    for id, read in id_read:
+        helpers.value_test(
+            ret_np_from_gt_flat=ta.read(id).numpy().flatten(),
+            ret_np_flat=np.array(ta_frontend.read(id)).flatten(),
+            backend=backend_fw,
+        )
+
+
+@given(l_kwargs=_helper_random_tensorarray())
+def test_tensorarray_size(
+    l_kwargs,
+    backend_fw,
+):
+    ta, ta_frontend = _helper_init_tensorarray(backend_fw, l_kwargs)
+    id_read, _ = l_kwargs
+    helpers.value_test(
+        ret_np_from_gt_flat=ta.size().numpy().flatten(),
+        ret_np_flat=np.array(ta_frontend.size()).flatten(),
+        backend=backend_fw,
+    )
+
+
+@given(
+    kwargs_v_l=_helper_tensorarray_split(),
+)
+def test_tensorarray_split(kwargs_v_l, backend_fw):
+    ta, ta_frontend = _helper_init_tensorarray(backend_fw, kwargs_v_l, "split")
+    for id in range(ta.size()):
+        helpers.value_test(
+            ret_np_from_gt_flat=ta.read(id).numpy().flatten(),
+            ret_np_flat=np.array(ta_frontend.read(id)).flatten(),
+            backend=backend_fw,
+        )
+
+
+@given(l_kwargs=_helper_random_tensorarray(fn="stack"))
+def test_tensorarray_stack(
+    l_kwargs,
+    backend_fw,
+):
+    ta, ta_frontend = _helper_init_tensorarray(backend_fw, l_kwargs)
+    helpers.value_test(
+        ret_np_from_gt_flat=ta.stack().numpy().flatten(),
+        ret_np_flat=np.array(ta_frontend.stack()).flatten(),
+        backend=backend_fw,
+    )
+
+
+@given(l_kwargs=_helper_tensorarray_unstack())
+def test_tensorarray_unstack(
+    l_kwargs,
+    backend_fw,
+):
+    ta, ta_frontend = _helper_init_tensorarray(backend_fw, l_kwargs, "unstack")
+    helpers.value_test(
+        ret_np_from_gt_flat=ta.stack().numpy().flatten(),
+        ret_np_flat=np.array(ta_frontend.stack()).flatten(),
+        backend=backend_fw,
+    )
+
```

Comment:
I know we are already using it everywhere, but still it would be nice to add a dedicated test for `write`.

Reply:
Yes so actually, `read` and `write` both will have exact same tests, I guess it would be better to add a comment rather than copying the same `read` test or maybe changing the name of test to `test_tensorarray_write_read`